### PR TITLE
Support http PATCH method

### LINKED ISSUE #52: Har files containing PATCH http requests make transformer crash
**Describe the bug**
har files containing patch requests fail

**Transformer version**
1.1.3

**To Reproduce**
Steps and input files to reproduce the behavior:
1. Create a har file with some web site using a REST api and update some field. REST uses patch http requests to do it. Attached [prueba3.zip](https://github.com/zalando-incubator/Transformer/files/3135015/prueba3.zip) as an example.

2. Run Transformer with the same command line explained in the examples
3. See error
```
  File ".../python3.6/site-packages/transformer/request.py", line 150, in from_har_entry
    method=HttpMethod[request["method"]],
  File "/usr/lib/python3.6/enum.py", line 329, in __getitem__
    return cls._member_map_[name]
KeyError: 'PATCH'
```

**Expected behavior**
A python available for locust should be produced

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Version:18.04

**Additional context**

### PR DESCRIPTION
**Pull request by jredrejo**
This PR adds PATCH method support to Transform

Closes #52

## Description

A simple code change to allow PATCH requests from har files being converted into proper Locust requests.

## Types of Changes

_What types of changes does your code introduce? Keep the ones that apply:_

- Bug fix (non-breaking change which fixes an issue)
## TODO

- [x] TODO 1

## Review

_Reviewers' checklist:_

- If this PR _implements_ new flows or _changes_ existing ones, are there
  **good tests** for these flows?
  If this PR rather _removes_ flows, are the obsolete tests removed as well?
- Is the documentation still up-to-date and exhaustive? This covers both
  _technical_ (in source files) and _functional_ (under `docs/`) documentation.
- Is the **changelog** updated?
- Does the **new version number** correspond to the actual changes from this PR?
  In doubt, refer to https://semver.org.

## After this PR

Any follow-up action necessary?

### CODE REVIEW

File: transformer/request.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ class HttpMethod(enum.Enum):
     PUT = enum.auto()  #: PUT
     OPTIONS = enum.auto()  #: OPTIONS
     DELETE = enum.auto()  #: DELETE
+    PATCH = enum.auto()  #: DELETE
```
Comment (bmaher) [inline transformer/request.py]: The doc. comment here should read `#: PATCH` (also makes sense to put this new enum right after `PUT`).

File: transformer/task.py [hunk 2]
```diff
@@ -212,7 +212,7 @@ def req_to_expr(r: Request) -> py.FunctionCall:
         if r.post_data:
             rpd = RequestsPostData.from_har_post_data(r.post_data)
             args.update(rpd.as_kwargs())
-    elif r.method is HttpMethod.PUT:
+    elif r.method in (HttpMethod.PUT, HttpMethod.PATCH):
```
Comment (bmaher) [inline transformer/task.py]: This should also be added to the equivalent block in the `lreq_to_expr` function (line 231).

File: transformer/request.py (hunk 1 above)
Comment (jredrejo) [reply transformer/request.py]: oops, done.

File: transformer/task.py (hunk 2 above)
Comment (jredrejo) [reply transformer/task.py]: Done !

**Comment by jredrejo** [comment]
> Thanks for your submission! I've posted a couple of small changes that need to be done. Also, would you mind adding a couple of tests to `test_task.py` as well? They should be similar to the tests for `PUT`.
> And finally, please update the changelog and version number as described here: https://transformer.readthedocs.io/en/latest/Contributing.html#suggested-workflow (read from step 9 onwards).
> Thanks again :)

Thanks for your guidance. I apologize for not having read in depth the contributor guidelines. Now I have done it ;) Congrats for the documentation, it's really useful and complete.

File: transformer/test_task.py [hunk 3]
```diff
@@ -360,6 +360,57 @@ def test_it_supports_put_requests_without_payload(self):
...
         )
 
+    def test_it_supports_patch_requests_with_payload(self):
+        url = "http://abc.de"
+        r = Request(
+            timestamp=MagicMock(),
+            method=HttpMethod.PATCH,
+            url=urlparse(url),
+            har_entry={"entry": "data"},
+            headers=[Header("a", "b")],
```
Comment (tortila) [inline transformer/test_task.py]: Version 1.2.0, which was released today, introduced a breaking change - headers are now stored in [`CaseInsensitiveDict`](https://github.com/zalando-incubator/Transformer/blob/master/transformer/request.py#L120), so there's no `Header` class anymore (hence the failing tests as reported in Travis). For the sake of readability it's fine to use a regular Python `dict` in the tests (like in other test cases in this module). 

Thanks for adding the tests! Have you considered adding your username to our [Contributors list](https://github.com/zalando-incubator/Transformer/blob/master/docs/Contributors.rst)? As a maintainer of this project I encourage you to do so 😄

Comment (jredrejo) [reply transformer/test_task.py]: Oops, I see I was so 'lucky' to commit almost at the same time 1.2.0 was merged into master.
I have fixed the tests and added myself to the contributors. Thanks again for your help!

**Pull request merged**